x/shield/types/types.go

```diff
@@ -6,20 +6,43 @@ import (
 	sdk "github.com/cosmos/cosmos-sdk/types"
 )
 
+// Pool contains a shield pool's data.
 type Pool struct {
-	PoolID          uint64         `json:"pool_id" yaml:"pool_id"`
-	Active          bool           `json:"active" yaml:"active"`
-	Description     string         `json:"description" yaml:"description"`
-	Sponsor         string         `json:"sponsor" yaml:"sponsor"`
-	SponsorAddr     sdk.AccAddress `json:"sponsor_address" yaml:"sponsor_address"`
-	Premium         MixedDecCoins  `json:"premium" yaml:"premium"`
-	TotalCollateral sdk.Int        `json:"total_collateral" yaml:"total_collateral"`
-	Available       sdk.Int        `json:"available" yaml:"available"`
-	TotalLocked     sdk.Int        `json:"total_locked" yaml:"total_locked"`
-	Shield          sdk.Coins      `json:"shield" yaml:"shield"`
-	EndTime         time.Time      `json:"end_time" yaml:"end_time"`
+	// PoolID is the id of the pool.
+	PoolID uint64 `json:"pool_id" yaml:"pool_id"`
+
+	// Description is the term of the pool.
+	Description string `json:"description" yaml:"description"`
+
+	// Sponsor is the project owner of the pool.
+	Sponsor string `json:"sponsor" yaml:"sponsor"`
+
+	// SponsorAddr is the CertiK Chain address of the sponsor.
+	SponsorAddr sdk.AccAddress `json:"sponsor_address" yaml:"sponsor_address"`
+
+	// Active means new purchases are allowed.
+	Active bool `json:"active" yaml:"active"`
+
+	// Premium is the undistributed pool
+	Premium MixedDecCoins `json:"premium" yaml:"premium"`
+
+	// TotalCollateral is the amount of all collaterals in the pool.
+	TotalCollateral sdk.Int `json:"total_collateral" yaml:"total_collateral"`
+
+	// Available is the amount of collaterals available to be purchased.
+	Available sdk.Int `json:"available" yaml:"available"`
+
+	// TotalLocked is the amount of
+	TotalLocked sdk.Int `json:"total_locked" yaml:"total_locked"`
+
+	// Shiled is the amount of unused shield for the pool sponsor.
```

Comment:
`unused shield` --> `purchased shield`

Reply:
How do we prevent B from getting over-reimbursement?

Reply:
We set purchases for B each time when we create or update pools. The amount of reimbursement will not exceed the purchase amount.

Reply:
Oh I misunderstood your comment. `unused shield` makes sense.